src/agent/src/rpc.rs

```diff
@@ -2101,18 +2120,34 @@ async fn cdh_handler(oci: &mut Spec) -> Result<()> {
         .ok_or_else(|| anyhow!("Spec didn't contain mounts field"))?;
 
     for m in mounts.iter_mut() {
-        if m.destination().starts_with("/sealed") {
-            info!(
-                sl(),
-                "sealed mount destination: {:?} source: {:?}",
-                m.destination(),
-                m.source()
-            );
-            if let Some(source_str) = m.source().as_ref().and_then(|p| p.to_str()) {
-                cdh::unseal_file(source_str).await?;
-            } else {
-                warn!(sl(), "Failed to unseal: Mount source is None or invalid");
+        if let Some(source_path) = m.source().as_ref().and_then(|p| p.to_str()) {
+            // Check if source_path starts with "/run/kata-containers/shared/containers"
+            // For a volume mount path /mydir,
+            // the secret file path will be like this under the /run/kata-containers/shared/containers dir
+            // a128482812bad768f404e063f225decd425fc94a673aec4add45a9caa1122ccb-75490e32e51da3ff-mydir
+            // We can ignore few paths like: resolv.conf, termination-log, hostname,hosts,serviceaccount
+            if should_process_source_path(source_path) {
+                info!(
```

Comment:
@ChengyuZhu6 I'm assuming your comment "The unrelated file to be unsealed may lead to confusion in logging." relates to this line. I can change it to debug (similar to the code in cdh.rs) to avoid confusion. Does that work?

Reply:
@ChengyuZhu6 I have changed it to debug. PTAL